app/client/websocket/index.ts

```diff
@@ -165,34 +160,33 @@ export default class WebSocketClient {
                 this.sendMessage('authentication_challenge', {token: this.token});
             }
 
-            if (this.connectFailCount > 0) {
+            if (this.shouldSkipSync) {
+                logInfo('websocket connected to', this.url);
+                this.firstConnectCallback?.();
+            } else {
                 logInfo('websocket re-established connection to', this.url);
                 if (!reliableWebSockets && this.reconnectCallback) {
                     this.reconnectCallback();
                 } else if (reliableWebSockets) {
+                    // If a sync is needed, it is handled when receiving the HELLO websocket message
                     this.reliableReconnectCallback?.();
                     if (this.serverSequence && this.missedEventsCallback) {
                         this.missedEventsCallback();
                     }
-                    this.hasReliablyReconnect = true;
                 }
-            } else if (this.firstConnectCallback) {
-                logInfo('websocket connected to', this.url);
-                this.firstConnectCallback();
             }
 
             this.connectFailCount = 0;
         });
 
         this.conn!.onClose(() => {
-            const now = Date.now();
-            if (this.lastDisconnect < this.lastConnect) {
-                this.lastDisconnect = now;
-            }
-
+            clearTimeout(this.connectionTimeout);
             this.conn = undefined;
             this.responseSequence = 1;
-            this.hasReliablyReconnect = false;
+
+            // If we close for whatever reason, we want to sync when we connect again
+            // unless said otherwise at initialize.
+            this.shouldSkipSync = false;
```

Comment:
Not really, sometimes you close the WS for a short period of time while switching apps and there is no real need to resync unless the server expired the connectionId

Reply:
Maybe I can improve the wording of the comment. The idea is that we want to sync when we connect again... to a new connection. If we have reliable websockets and we reconnect to the same connection, we don't re-sync, we just ignore this.